packages/jest-core/src/TestScheduler.ts

```diff
@@ -330,25 +332,40 @@ class TestScheduler {
     }
   }
 
-  private _shouldAddDefaultReporters(
-    reporters?: Array<string | Config.ReporterConfig>,
-  ): boolean {
-    return (
-      !reporters ||
-      !!reporters.find(
-        reporter => this._getReporterProps(reporter).path === 'default',
-      )
-    );
-  }
-
   async _setupReporters() {
     const {collectCoverage, notify, reporters} = this._globalConfig;
-    const isDefault = this._shouldAddDefaultReporters(reporters);
+
+    if (notify) {
+      this.addReporter(
+        new NotifyReporter(
+          this._globalConfig,
+          this._options.startRun,
+          this._context,
+        ),
+      );
+    }
+
+    if (!reporters) {
+      this._setupDefaultReporters(collectCoverage);
+      return;
+    }
```

Comment:
Adding `NotifyReporter` first, allows to return early. Also later `reporters &&` checks become unnecessary.